gms/gossiper.cc

```diff
@@ -1327,14 +1327,16 @@ future<> gossiper::assassinate_endpoint(sstring address) {
     });
 }
 
-future<generation_type> gossiper::get_current_generation_number(inet_address endpoint) {
-    return container().invoke_on(0, [endpoint] (auto&& gossiper) {
+future<generation_type> gossiper::get_current_generation_number(inet_address endpoint) const {
+    // FIXME: const container() has no const invoke_on variant
+    return const_cast<gossiper*>(this)->container().invoke_on(0, [endpoint] (const gossiper& gossiper) {
```

Comment:
Why do you need get_current_generation_number() to be const?

Reply:
No need other than to accurately reflect what it does, or does not in this case.

Reply:
Will be deferred to a followup series

Reply:
What I'm worried about is that marking things as const adds complications like in this FIXME, duplication of logic between const and non-const variants, and can put impediments to future extension when need to mutate something. So to pull its weight, it should be balanced by benefits in the form of more safety. In case of a gossiper, do we need a concept of constness? Maybe we don't.

Reply:
At an earlier version of this branch, this helped me sanitize the gossiper to make sure some const functions are guaranteed not to change _endpoint_state_map.

The const_cast here unfortunately was required due to limitation of our `container().invoke_on` templates that require a mutable instance for invoke_on.
I spent some time trying to come up with const overrides of these template, but I got tangled in the c++ templates web and decided to drop this effort :-/